Repository/AbstractPhpcrRepository.php

```diff
@@ -147,6 +148,15 @@ public function add($path, $resource)
     }
 
     /**
+     * {@inheritdoc}
+     */
+    public function reorder($sourcePath, $position)
+    {
+        Assert::greaterThanEq($position, 0, 'Reorder position cannot be negative, got: %s');
+        $this->reorderNode($sourcePath, $position);
```

Comment:
Well, we could mark the implementations (e.g. OdmRepository) as final. Which covers us in the
future.

On Mon, Jun 27, 2016 at 01:48:38AM -0700, David Buchmann wrote:
>    In [1]Repository/AbstractPhpcrRepository.php:
> 
>  > @@ -147,6 +148,15 @@ public function add($path, $resource)
>  >      }
>  >
>  >      /**
>  > +     * {@inheritdoc}
>  > +     */
>  > +    public function reorder($sourcePath, $position)
>  > +    {
>  > +        Assert::greaterThanEq($position, 0, 'Reorder position cannot be negative, got: %s');
>  > +        $this->reorderNode($sourcePath, $position);
> 
>    being a protected method, somebody could still extend the abstract
>    repository and do custom stuff, so changing the name once its released is
>    a BC break. but i can live with having Node in the name, especially as its
>    consistent with the rest of the class. (that the odm uses nodes underneath
>    is no argument imho, its behind an abstraction layer.)
> 
>    —
>    You are receiving this because you authored the thread.
>    Reply to this email directly, [2]view it on GitHub, or [3]mute the thread.
> 
>    Reverse link: [4]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/symfony-cmf/resource/pull/28#discussion_r68541600
>    2. https://github.com/symfony-cmf/resource/pull/28/files/6bb50cfa74bf87e110e9b44c5c4a639b98ef8f37#r68541600
>    3. https://github.com/notifications/unsubscribe/AAgZcdjrGQHwId70zBmaor47IVbZznR_ks5qP45mgaJpZM4I9hWA
>    4. https://github.com/symfony-cmf/resource/pull/28/files/6bb50cfa74bf87e110e9b44c5c4a639b98ef8f37#r68541600